### PR TITLE
WIP: add a 3rd argument to Huber function

### PR DESCRIPTION
**Pull request by CSHugoM**
I changed the huber function to add a 3rd argument T which represents the possible scaling of the function.

### CODE REVIEW

**Comment by rileyjmurray** [comment]
Thanks so much for giving this a shot, @CSHugoM! Could you please add some tests to verify that it works as expected?

As a starting point, I suggest a test where you minimize this function where the input vectors and scale parameter are confined to a simple box in the positive orthant (say, all components in the interval [1, 2]). You should be able to work out an analytic solution for that problem. I also suggest looking at an existing test with the two-parameter Huber function, using a Variable ``scale`` input, but fixing ``scale == 1``. Then you can check that the behavior is unchanged relative to the two-parameter Huber function.

(Also, sorry for not replying to your GitHub issue. I know I directed you here, and we really appreciated your enthusiasm.)

**Comment by CSHugoM** [comment]
Thanks for your comments !

I wrote the first test in the test_problem.py file. However, when I try to run it with the command : 
python -m pytest test_problem.py::TestProblem::test_huber_3_arg_min

I have an Error : Not Found.

Have you any idea why ? 

Thanks,
Hugo

File: cvxpy/atoms/elementwise/huber.py [hunk 1]
```diff
@@ -43,17 +43,20 @@ class huber(Elementwise):
...
         self.M = self.cast_to_const(M)
+        self.T = self.cast_to_const(T)
         super(huber, self).__init__(x)
 
     @Elementwise.numpy_numeric
     def numeric(self, values) -> float:
         """Returns the huber function applied elementwise to x.
         """
-        return 2*scipy.special.huber(self.M.value, values[0])
+        return 2*scipy.special.huber(self.M.value, values[0]) if self.T.value == 0 else self.T.value*(1 + 2*scipy.special.huber(self.M.value/self.T.value, values[0]))
```
Comment (SteveDiamond) [inline cvxpy/atoms/elementwise/huber.py:59]: This line is way too long. Break it out into a proper if statement.

Comment (CSHugoM) [reply cvxpy/atoms/elementwise/huber.py:59]: Thanks for pointing this out, I'll take break it out then.

**Comment by rileyjmurray** [comment]
@CSHugoM I left a comment on the GitHub issue that provides more guidance. Take a look and let me know (on that thread) if those resources don't make total sense.

**Comment by CSHugoM** [comment]
@rileyjmurray Thanks for that ! Are you talking about Issue #1377 ? I don't see any new comment on this issue, is this about one of your old comments ?

**Comment by rileyjmurray** [comment]
Huh, I swear left a comment, but it's not there. I just added one again.

**Comment by rileyjmurray** [comment]
@CSHugoM do you feel like you have a handle on what needs to be done? If not, maybe we could have a Skype / Zoom call.

**Comment by rileyjmurray** [comment]
Closing due to inactivity.

**Pull request closed without merging**